website/www/check-links.sh

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+set -euo pipefail
+
+
+function redraw_progress_bar { # int barsize, int base, int current, int top
```

Comment:
Do we really need this graphical thing? Most of these runs will be in jenkins.

Reply:
I think it's good to know the progress when someone wanna run the test locally `./gradlew website:testWebsite` 😄

Reply:
How long does it take to run it?

Reply:
Around 20-30 mins, since it gets the status response from all of the links (approx. 1500 if I don't remember wrong).

Reply:
This script should be very parallelizable since it's completely IO bound. Maybe consider something like `xargs -P` rather than a bash for loop